util/runtime/panic.go

```diff
@@ -0,0 +1,20 @@
+package runtime
+
+import (
+	"runtime"
+
+	log "github.com/sirupsen/logrus"
+)
+
+func RecoverFromPanic(log *log.Entry) {
+	if r := recover(); r != nil {
+		// Same as stdlib http server code. Manually allocate stack trace buffer size
+		// to prevent excessively large logs
+		const size = 64 << 10
+		stacktrace := make([]byte, size)
+		stacktrace = stacktrace[:runtime.Stack(stacktrace, false)]
```

Comment:
I would recommend splitting this into multiple lines for easier readability. Not clear here that `runtime.Stack(stacktrace, false)` returns the number of bytes written.